src/app/domain/navigation/webviews/UrlService.ts

```diff
@@ -0,0 +1,180 @@
+}
+
+export const interceptNavigation = async ({
+  initialRequest,
+  targetUri,
+  subdomainType,
+  navigation,
+  onShouldStartLoadWithRequest,
+  interceptReload,
+  onReloadInterception,
+  isFirstCall,
+  client,
+  setDownloadProgress
+}: InterceptNavigationProps): Promise<boolean> => {
+  const isPreviewableLink = checkIsPreviewableLink(initialRequest.url, client)
+  // We don't want to intecerpt iframe navigation excepts for iOS and the download
+  // Since we can download file from an iframe (intents)
+  if (
+    Platform.OS === 'ios' &&
+    !initialRequest.isTopFrame &&
+    !isPreviewableLink
+  ) {
+    return true
+  }
+
+  if (interceptReload) {
+    const preventRefreshByDefault = isFirstCall
+    const isReload = checkIsReload(initialRequest, preventRefreshByDefault)
+
+    if (isReload) {
+      webviewUrlLog.debug('Intercepting reload, remount component instead')
+      onReloadInterception()
+      return false
+    }
+  }
+
+  if (isPreviewableLink) {
+    const fileExtension = getFileExtentionFromCozyDownloadUrl(
+      initialRequest.url,
+      isPreviewableLink
+    )
+    if (Platform.OS === 'ios' || fileExtension === 'pdf') {
+      safePromise(previewFileFromDownloadUrl)({
+        downloadUrl: initialRequest.url,
+        client,
+        setDownloadProgress
+      })
+      return false
+    } else {
+      return true
+    }
+  }
+
+  const newSlug = checkIsSlugSwitch({
+    currentUrl: targetUri,
+    destinationUrl: initialRequest.url,
+    subdomainType
+  })
+
+  if (newSlug) {
+    try {
+      await navigateToApp({
+        navigation,
+        href: initialRequest.url,
+        slug: newSlug
+      })
+    } catch (error) {
+      webviewUrlLog.warn('Could not navigate to app', error)
+    }
+
+    return false
+  }
+
+  if (_isNotHttpOrHttps(initialRequest.url)) {
+    safePromise(_handleNonHttpOrHttps)(initialRequest.url)
+    return false
+  }
+
+  const isRedirectOutside = checkIsRedirectOutside({
+    currentUrl: targetUri,
+    destinationUrl: initialRequest.url
+  })
+
+  if (isRedirectOutside) {
+    openUrlInAppBrowser(initialRequest.url)
+    return false
+  }
+
+  return onShouldStartLoadWithRequest(initialRequest)
+}
+
+export const interceptOpenWindow = async ({
+  currentUrl,
+  destinationUrl,
+  subdomainType,
+  navigation,
+  webViewForwardRef
+}: InterceptOpenWindowProps): Promise<void> => {
+  const isSameApp = checkIsSameApp({
+    currentUrl,
+    destinationUrl,
+    subdomainType
+  })
+
+  if (isSameApp) {
+    navigateTo(webViewForwardRef, destinationUrl)
+    return
+  }
+
+  const newSlug = checkIsSlugSwitch({
+    currentUrl,
+    destinationUrl,
+    subdomainType
+  })
+
+  if (newSlug) {
+    try {
+      await navigateToApp({
+        navigation,
+        href: destinationUrl,
+        slug: newSlug
+      })
+    } catch (error) {
+      webviewUrlLog.warn('Could not navigate to app', error)
+    }
+
+    return
+  }
+
+  if (_isNotHttpOrHttps(destinationUrl)) {
```

Comment:
Why are you using the `_XXX` notation?

Reply:
If you rename the method `openUrlWithOS()` or similar, then it is meant to be exported and can be easily used by other files 😛

Reply:
👍